### ISSUE TITLE
azure-cli-core dependency

### ISSUE THREAD

**Issue by vizhur** [Azure/shrike#234]
from Azure SDK:

Azure CLI do not consider itself as a Python package anymore, and the PyPI scenario has been removed from official installation documentation (How to install the Azure CLI | Microsoft Docs). I’m actually surprised it’s still pushed on PyPI, I thought it has been stopped, but ****** can provide more detail if installing any CLI part from PyPI is a good or bad idea. As far as I know, there is no process in place to guarantee any backward compatibility of the Python layer, only the external CLI one. And don’t get me wrong, I think this is a good choice, we should see CLI as an application and the coding language is an implementation detail. From that point of view, you should get rid of usage of “azure-cli-core” entirely in the first place as a Python dependency.

besides having that dependency was always a bad idea, esp if you just rely on nested dependencies.

other argument, that probably will go into conflict if run in Azure CLi task in ADO

**Comment by dkmiller** [comment]
Related: https://github.com/Azure/azure-cli/issues/20150 .

**Comment by vizhur** [comment]
> Related: [Azure/azure-cli#20150](https://github.com/Azure/azure-cli/issues/20150) .

questionably related :), that is a general feedback for the dependency and some guidance from Azure SDK team (https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)

**Comment by haozhestat** [comment]
@vizhur and @dkmiller : If we remove `azure-cli-core` from the dependency list of shrike, we will get the following error:

```
"message": "azure.cli.core package is not installed. AzureCliAuthentication requires azure.cli.core to be installed in the same python environment where azureml-sdk is installed."
```

The reason is that `AzureCliAuthentication` (`from azureml.core.authentication import AzureCliAuthentication`) depends on `azure.cli.core` **implicitly**.

https://github.com/Azure/shrike/blob/be7c1dbcc11d7fac1ded760b7c64697b80ae86cb/shrike/pipeline/aml_connect.py#L137

**Comment by vizhur** [comment]
agree, that is sadness you need to complain to AzureML Python SDK team if you do not have other solutions for auth

**Comment by fuhuifang** [comment]
Suggestion from Hanchi: use https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.azureclicredential?view=azure-python
